src/agoradatatools/etl/transform/model_details.py

```diff
@@ -213,6 +213,12 @@ def transform_model_details(
     model_info_df = datasets["model_info"].fillna("")
     human_transgene_allele_map_df = datasets["human_transgene_allele_map"].fillna("")
 
+    # Ensure jax_id preserves leading zeros by converting to string with proper formatting
```

Comment:
Nit: reading this comment made me think that the input data was as string with leading 0's but really this is a transform to append leading 0s if `jax_id` is not null, is that correct?

Also -  is it ok for `jax_id` to be null?